src/main/java/edu/greenblitz/tobyDetermined/commands/telescopicArm/moveByNodes/StraightLineMotion.java

```diff
@@ -0,0 +1,146 @@
+import edu.greenblitz.tobyDetermined.RobotMap;
+import edu.greenblitz.tobyDetermined.subsystems.telescopicArm.ArmSimulation;
+import edu.greenblitz.tobyDetermined.subsystems.telescopicArm.Elbow.Elbow;
+import edu.greenblitz.tobyDetermined.subsystems.telescopicArm.Extender.Extender;
+import edu.greenblitz.utils.GBMath;
+import org.littletonrobotics.junction.Logger;
+
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeBase.CreateCurrents.system1CurrentNode;
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeBase.CreateCurrents.system1MidNode;
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeSystemUtils.getNode;
+import static edu.greenblitz.tobyDetermined.RobotMap.TelescopicArm.Extender.*;
+
+
+public class StraightLineMotion extends GBCommand {
+    private final Extender extender;
+    private final Elbow elbow;
+    private final SystemsState startState;
+    private final SystemsState endState;
+    private final NodeArm startingNode;
+    private final NodeArm endNode;
+    private double combinedVelocity;
+    private static final double STATIC_COMBINED_VELOCITY = 1; //Meters Per Second
+    private static final double MAX_EXTENDER_VELOCITY = 1.5; //Meters Per Second
+    private static final double MAX_ANGULAR_VELOCITY = 3;//Radians Per Second
+    private static final double TO_BELLY_VELOCITY = 0.2;
+    private static final double ONLY_ELBOW_VELOCITY = 1;
+
+    public StraightLineMotion(SystemsState start, SystemsState end) {
+        extender = Extender.getInstance();
+        elbow = Elbow.getInstance();
+        require(elbow);
+        require(extender);
+        combinedVelocity = 0;
+        this.startState = start;
+        this.endState = end;
+        startingNode = (NodeArm) getNode(start);
+        endNode = (NodeArm) getNode(end);
+    }
+
+    @Override
+    public void initialize() {
+        if (extender.getLength() + endNode.getExtendPosition() <= DISTANCE_NOT_STRAIGHT_LINE)
+            combinedVelocity = ONLY_ELBOW_VELOCITY;
+
+        else if (endState.equals(SystemsState.ARM_GROUND))
+            combinedVelocity = TO_BELLY_VELOCITY;
+
+        else {
+            double length1 = startingNode.getExtendPosition() + STARTING_LENGTH;
+            double length2 = endNode.getExtendPosition() + STARTING_LENGTH;
+            double distance = GBMath.distance(
+                    GBMath.convertToCartesian(length1, startingNode.getAnglePosition()),
+                    GBMath.convertToCartesian(length2, endNode.getAnglePosition())
+            );
+            combinedVelocity = STATIC_COMBINED_VELOCITY * GBMath.sigmoid(distance, 1, 5.9, 0.3);//numbers that work well
+        }
+
+        Logger.getInstance().recordOutput("Arm/TargetPose3D/endNode", ArmSimulation.getArmPosition(endNode.getExtendPosition(), (endNode.getAnglePosition())));
+    }
+
+    public static double getRatioBetweenAngleAndLength(double sideA, double sideB, double gamma) {
```

Comment:
Could this be moved outside of the class? Doesn't seem relevant to the command itself and the file is kinda long

In general, I see a few functions that are a bit misplaced

Reply:
GBMath?